src/glmfit.jl

```diff
@@ -26,72 +26,104 @@ type GlmResp{V<:FPVector,D<:UnivariateDistribution,L<:Link} <: ModResp       # r
         lo = length(off)
         lo == 0 || lo == n || error("offset must have length $n or length 0")
         res = new(y,d,l,similar(y),eta,mu,similar(y),off,similar(y),wts,similar(y),similar(y))
-        updatemu!(res, eta)
+        updateμ!(res, eta)
         res
     end
 end
 
 # returns the sum of the squared deviance residuals
 deviance(r::GlmResp) = sum(r.devresid)
 
-function updatemu!{T<:FPVector}(r::GlmResp{T}, linPr::T)
-    y = r.y
-    dist = r.d
-    link = r.l
-    eta = r.eta
-    mu = r.mu
-    muetav = r.mueta
-    offset = r.offset
-    var = r.var
-    wts = r.wts
-    wrkresid = r.wrkresid
-    devresidv = r.devresid
+function wtscale!{T<:FPVector}(devr::T, wkwt::T, wt::T)
+    if !isempty(wt)
+        for i in eachindex(devr, wkwt, wt)
+            wti = wt[i]
+            devr[i] *= wti
+            wkwt[i] *= wti
+        end
+    end
+end
+
+
+function updateμ!{T<:FPVector, D<:Union{Bernoulli, Binomial}, L<:LogitLink}(
+    m::GlmResp{T,D,L}, η::T)
+
+    y, μ, wrkres, wrkwt, dres = m.y, m.mu, m.wrkresid, m.wrkwts, m.devresid
+
+    @inbounds Threads.@threads for i in 1:length(μ)
```

Comment:
Maybe add `using Base.Threads` at the top to avoid repeating `Threads.`? Really cool that we can use several cores!